au/apply_magnitude.hh

```diff
@@ -141,13 +164,12 @@ struct ApplyMagnitudeImpl<Mag, ApplyAs::RATIONAL_MULTIPLY, T, true> {
                   "Mismatched instantiation (should never be done manually)");
 
     constexpr T operator()(const T &x) {
-        return x * get_value<T>(numerator(Mag{})) / get_value<T>(denominator(Mag{}));
+        return static_cast<T>(x * get_value<T>(numerator(Mag{})) /
```

Comment:
Commend: Nice catch!

Reply:
Actually, in chatting about the logic with you, I noticed a bug, and it was here!  Fortunately, it was pretty self-contained, and easy to write a test case for.  Check out 81c4a3bcd1f4da4b9f3d89019d77119228dc3b57 for the fix.

Reply:
Great! I was thinking about `static_cast<T>`. I guess there's also the tricky bit about `get_value<P>`. This line is trickier than it looks 😅